File: Batch12/gaurav/array/PlusOne.java
class Solution {
    public int[] plusOne(int[] digits) {
        
        int n = digits.length - 1;

        while(n >= 0) {
            if(digits[n] == 9) {
                digits[n] = 0;
            } else {
                
                digits[n] = digits[n]+1;
                return digits;
            }
            n--;
        }

        int[] digits1 = new int[digits.length+1];
        for(int i=n; i>0; i--) {
            digits1[i+1] = digits[i]; 
        }

        digits1[0] = 1;
        digits1[1] = 0;
        return digits1;
    }
}

File: B13/Subhendu/Array/plus-one.py
class Solution:
    def plusOne(self, digits: List[int]) -> List[int]:
        ## https://leetcode.com/problems/plus-one/

        i = len(digits) - 1

        while i >= 0:
            if digits[i] != 9:
                digits[i] += 1
                return digits
            else:
                digits[i] = 0
            i -= 1

        return [1] + digits